uniffi_macros/src/error.rs

```diff
@@ -143,24 +197,39 @@ fn error_metadata(
 
 mod kw {
     syn::custom_keyword!(flat_error);
+    syn::custom_keyword!(with_try_read);
 }
 
 #[derive(Default)]
 struct ErrorAttr {
     flat: Option<kw::flat_error>,
+    with_try_read: Option<kw::with_try_read>,
 }
 
 impl Parse for ErrorAttr {
     fn parse(input: ParseStream<'_>) -> syn::Result<Self> {
-        let flat = input.parse()?;
-        Ok(ErrorAttr { flat })
+        let lookahead = input.lookahead1();
+        if lookahead.peek(kw::flat_error) {
+            Ok(Self {
+                flat: input.parse()?,
+                ..Self::default()
+            })
+        } else if lookahead.peek(kw::with_try_read) {
+            Ok(Self {
+                with_try_read: input.parse()?,
+                ..Self::default()
+            })
+        } else {
+            Err(lookahead.error())
+        }
```

Comment:
This is my best guess at how to handle more attributes here, please tell me if there's a better way to do this.

Reply:
Exactly what I would have written 👍🏼